pkg/virt-api/rest/authorizer.go

```diff
@@ -27,11 +27,10 @@ import (
 	"net/http"
 	"strings"
 
-	restful "github.com/emicklei/go-restful"
-	authorization "k8s.io/api/authorization/v1"
+	"github.com/emicklei/go-restful"
+	authv1 "k8s.io/api/authorization/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	authorizationclient "k8s.io/client-go/kubernetes/typed/authorization/v1"
-	restclient "k8s.io/client-go/rest"
+	authclientv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
```

Comment:
This authorizer code is one of the highest risk codepaths we have, so we have to be very careful and critical of changes made here. Some of the changes in this pr seem to be cosmetic in nature, such as these import renames, which makes this harder to review with confidence We also have what looks like a high degree of refactoring occurring in both the authorizer code and the unit test.

I'd have a higher degree of confidence this code is accurate if the existing unit test cases were not modified at all (if possible) and only new cases where added to account for the namespace changes. I'd also like to see the authorizer.go file itself only modified where the change in behavior is occurring so we can look at these changes very critically. Basically, do as little as possible for this round.

Now, I want to be clear in saying refactoring all of this is welcome. I'd just like to see it in a standalone PR that we know isn't coupled with behavior changes as well... and again this is simply due to these code paths being of high risk.

Reply:
Fair enough, I will update the PR.